### PR TITLE
mu4e: replace obsolete functions

### PR DESCRIPTION
**Pull request by tshu-w**

### CODE REVIEW

File: modes/mu4e/evil-collection-mu4e.el [hunk 1]
```diff
@@ -155,14 +155,14 @@ with older release versions of `mu4e.'"
      "o" mu4e-headers-change-sorting
      "j" mu4e-headers-next
      "k" mu4e-headers-prev
-     "gr" mu4e-headers-rerun-search
-     "b" mu4e-headers-search-bookmark
-     "B" mu4e-headers-search-bookmark-edit
+     "gr" mu4e-search-rerun
```
Comment (condy0919) [inline modes/mu4e/evil-collection-mu4e.el:158]: Setting `revert-function-function` to `mu4e-search-return` in `mu4e-headers-mode` smells good, which let evil-collection maintain less keybindings. e.g. https://github.com/emacs-dashboard/emacs-dashboard/pull/352

If you would like to PR mu4e, 😀. Note, it's a bonus.

**Review by condy0919 — CHANGES_REQUESTED** [review]

File: modes/mu4e/evil-collection-mu4e.el [hunk 2]
```diff
@@ -259,7 +259,7 @@ with older release versions of `mu4e.'"
            (mu4e-headers-mark-thread nil '(read)))
      ,@(when evil-want-C-u-scroll
          '("\C-u" evil-scroll-up))))
-  ;; TODO: Add mu4e-headers-search-bookmark?
+  ;; TODO: Add mu4e-search-bookmark?
```
[the remark this answers is not in the document]
Comment (tshu-w) [reply modes/mu4e/evil-collection-mu4e.el]: Done.

File: modes/mu4e/evil-collection-mu4e.el (hunk 1 above)
Comment (tshu-w) [reply modes/mu4e/evil-collection-mu4e.el:158]: I have saved it in my inbox. Once my PR to mu merged, I will open a new PR.

**Comment by condy0919** [comment]
Looks good

**Comment by omajid** [comment]
This change breaks users who are using mu 1.6.x which still uses the older names. 1.6.11 was released on June 2022, and distributions like Fedora are still using mu 1.6.x.

**Pull request merged**